### PR TITLE
ovirt_host_networks: Fix of bond assignment

### PR DESCRIPTION
**Pull request by machacekondra**
##### SUMMARY

This patch fixes few scenarios for setup networks.

1) When mgmt interface is on `eth0` and we want to create bond of `eth0` and `eth1` assigned to mgmt network.
```
- name: Create bond on mgmt network
    ovirt_host_networks:
      auth: "{{ ovirt_auth }}"
      state: present
      name: host1
      bond:
        name: bond0
        mode: 1
        interfaces:
          - eth0
          - eth2
      networks:
        - name: ovirtmgmt
          boot_protocol: static
          address: 192.168.121.229
          netmask: 255.255.255.0
          gateway: 192.168.121.1
      save: true
      check: true

```

2) When mgmt interface is on `bond0` of `eth0` and `eth1` and we want to assign mgmt interface only on `eth0`.
```
- name: Assign mgmt network on eth0
    ovirt_host_networks:
      auth: "{{ ovirt_auth }}"
      name: "host1"
      interface: eth0
      networks:
        - name: ovirtmgmt
          boot_protocol: static
          address: 192.168.121.229
          netmask: 255.255.255.0
          gateway: 192.168.121.1
      check: true
      save: true
```

3) Fix bond mode assignment, previously we didn't pass `miimon=100` for `mode=1,2,3` and we didn't pass `xmit_hash_policy=2` for mode=4.

4) We incorrectly documented `prefix` for bond, instead of `netmask`, `netmask` always worked, so removing the `prefix` and replacing it with `netmask` in documentation`.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

ovirt_host_networks

##### ANSIBLE VERSION

```
2.5.0
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py [hunk 1]
```diff
@@ -57,7 +57,7 @@
             - "C(name) - Name of the logical network to be assigned to bond or interface."
             - "C(boot_protocol) - Boot protocol one of the I(none), I(static) or I(dhcp)."
             - "C(address) - IP address in case of I(static) boot protocol is used."
-            - "C(prefix) - Routing prefix in case of I(static) boot protocol is used."
+            - "C(netmask) - Netmask in case of I(static) boot protocol is used."
```
Comment (mwperina) [inline lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: "Netmask in case ..." -> "Subnet mask in case ..."

**Review by mwperina — CHANGES_REQUESTED** [review]
Small doc correction, other than that looks good

**Comment by mwperina** [comment]
@dankenigsberg Dan, could you please take a look?

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py (hunk 1 above)
Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: done

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py [hunk 2]
```diff
@@ -287,44 +334,41 @@ def main():
...
+            for a in attachments_service.list():
+                current_network_name = get_link_name(connection, a.network)
+                if current_network_name in network_names:
+                    for n in networks:
+                        if n['name'] == current_network_name:
+                            n['id'] = a.id
+
+            # Check if we have to break some bonds:
+            for host_nic in nics_service.list():
+                if host_nic.bonding and nic.id in [slave.id for slave in host_nic.bonding.slaves]:
```
Comment (knumskull) [inline lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: It could be possible, that `nic` is `None` and in this case, this check will fail

~~~
  File "/tmp/ansible_WEMKC_/ansible_module_ovirt_host_networks.py", line 311, in main
    if host_nic.bonding and nic.id in [slave.id for slave in host_nic.bonding.slaves]:
AttributeError: 'NoneType' object has no attribute 'id'
~~~

Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: Right thanks, in case there exist different bond and we are trying to create another which don't yet exist it may fail. Fixed.

**Review by mwperina — APPROVED** [review]
shipit

**Comment by dankenigsberg** [comment]
@mwperina, I've missed your review request. Does this Ansible module have a way to specify customer bonding options? If it does, I do not see it. If not, we need an RFE for it.

**Comment by machacekondra** [comment]
@dankenigsberg  It does, we need to set bond with eth0 and eth1 and assign `ovirtmgmt ` network on that bond, if the `ovirtmgmt` network is already assigned on eth0. This is what this PR fixes and few other issues..

**Comment by dankenigsberg** [comment]
@mwperina I don't understand your answer. How would a user specify a fancy set of bonding options, e.g. miimon=700 ?

**Comment by machacekondra** [comment]
@dankenigsberg Such feature isn't part of this PR. This is just a bug fix. If we want such feature we need to create another PR and target it to Ansible 2.6.
